keystone-shell/keystone_shell_vars.py

```diff
@@ -0,0 +1,152 @@
+KEYSTONE_VERIFY = os.environ.get('KEYSTONE_VERIFY', 'true') == 'true'
+KEYSTONE_CERT = os.environ.get('KEYSTONE_CERT', None)
+
+_kubernetes_client = None
+
+
+def get_current_namespace() -> str:
+    if 'NAMESPACE' in os.environ:
+        return os.environ['NAMESPACE']
+
+    if os.path.exists(NAMESPACE_FILE):
+        with open(NAMESPACE_FILE, 'r') as f:
+            return f.read()
+
+    logger.warning('Not running in cluster and $NAMESPACE is not set, '
+                   'assuming \'default\'!')
+    return 'default'
+
+
+def get_kubernetes_client() -> KubernetesAPIClient:
+    global _kubernetes_client
+
+    if _kubernetes_client is None:
+        _kubernetes_client = KubernetesAPIClient()
+        _kubernetes_client.load_auto_config()
+
+    return _kubernetes_client
+
+
+def keystone_env_vars():
+    ret = {}
+    for arg in KEYSTONE_PASSWORD_ARGS:
+        name = 'OS_{}'.format(arg.upper())
+        if name in os.environ:
+            ret[name] = os.environ[name]
+
+    return ret
+
+
+def keystone_args_from_env():
+    ret = {}
+    for arg in KEYSTONE_PASSWORD_ARGS:
+        ret[arg] = os.environ.get('OS_{}'.format(arg.upper()))
+
+    return ret
+
+
+def get_keystone_client():
+    auth = Password(**keystone_args_from_env())
+    session = Session(auth=auth,
+                      app_name='keystone-shell',
+                      user_agent='keystone-shell',
+                      timeout=KEYSTONE_TIMEOUT,
+                      verify=KEYSTONE_VERIFY,
+                      cert=KEYSTONE_CERT)
+
+    discover = Discover(session=session)
+    return discover.create_client()
+
+
+def get_kubernetes_secret(name: str,
+                          namespace: str=None) -> Union[KubernetesAPIResponse,
+                                                        None]:
+    """
+    :return: loaded secret dict or None if it does not exist
+    """
+    client = get_kubernetes_client()
+
+    if namespace is None:
+        namespace = get_current_namespace()
+
+    try:
+        return client.get('/api/v1/namespaces/{}/secrets/{}', namespace, name)
+    except HTTPError as e:
+        if e.response.status_code != 404:
+            raise
+
+        return None
+
+
+def main():
+    if len(sys.argv) > 1:
+        secret_name = sys.argv[1]
+
+        if len(sys.argv) > 2:
+            secret_namespace = sys.argv[2]
+        else:
+            secret_namespace = None
+
+        # purge existing keystone vars from the environment
+        for var in keystone_env_vars().keys():
+            print('unset {};'.format(var))
+
+        secret = get_kubernetes_secret(secret_name, secret_namespace)
+        for key, val in secret.data.items():
+            val_bytes = base64.b64decode(val)
+            print('export {}="{}"'.format(key, val_bytes.decode('utf-8')))
+
+        logger.info('now using account from secret %s', secret_name)
+    else:
+        logger.info('no secret name arg provided, will use default environment')
```

Comment:
Maybe you could move it up with reversed `if len(sys.argv) > 1:` to have it close to the place why did it run and not with just bare `else` after a lot of code.

Reply:
I think you're right, it's more clear to write that a bit differently.